api/v1/lib/extras/store/singleton.go

```diff
@@ -0,0 +1,157 @@
+package store
+
+import (
+	"errors"
+	"sync/atomic"
+)
+
+type (
+	Getter interface {
+		Get() (string, error)
+	}
+
+	GetFunc func() (string, error)
+
+	Setter interface {
+		Set(string) error
+	}
+
+	SetFunc func(string) error
+
+	// Singleton is a thread-safe abstraction to load and store a string
+	Singleton interface {
+		Getter
+		Setter
+	}
+
+	SingletonAdapter struct {
+		GetFunc
+		SetFunc
+	}
+
+	SingletonDecorator interface {
+		Decorate(Singleton) Singleton
+	}
+)
+
+func (f GetFunc) Get() (string, error) { return f() }
+func (f SetFunc) Set(s string) error   { return f(s) }
+
+var ErrNotFound = errors.New("value not found in store")
```

Comment:
not crazy about this. maybe a "not found" condition should just return empty string?